### ISSUE TITLE
More details on File Structure

### ISSUE THREAD

**Issue by jvitoroc** [reactjs/reactjs.org#330]
Some questions need to be answered, like "How the features are connected to each other" and "How to connect the features" best practices and bad practices
https://github.com/reactjs/reactjs.org/blob/master/content/docs/faq-structure.md

**Comment by bvaughn** [comment]
I think this might be a bit too broad.

How to connect "features" depends on the context. What kind of features (components) are you connecting? If they're pages, then you could use component `state` or a routing solution like React Router. If they're smaller elements (eg buttons) then you can just require/import them directly and use them within another component's `render` method.

This sort of thing can vary by preference and I don't think the team wants to be too prescriptive here. There are many good solutions for how to connect components/features/etc.

PRs are always welcome for adding clarity and improving the docs. If you have ideas, please feel free to share a proposal by posting a PR. I don't think this issue is particularly actionable as-is though, so I'm going to close it.

Please feel free to ask for clarification if any of what I've said is unclear.

**Comment by jvitoroc** [comment]
I'm having a hard time trying to develop this app: https://github.com/jvitoroc/issues-analyst/tree/ref
I don't know I'm following the best practices and if my file structure is good. Can you take a look?

**Comment by bvaughn** [comment]
Hey @jvitoroc. Dan added a bit more context to the [File Structure page](https://reactjs.org/docs/faq-structure.html) with PR #338. You might want to check it out again?

I don't really have the bandwidth to audit projects and give tailored feedback unfortunately. 🙁  I'm sorry.

**Comment by jvitoroc** [comment]
@bvaughn Thanks, but at least tell me if it's alright copy and paste the same component in every folder inside src/. In my project I do this with the Message Component.
Sorry for my english

**Comment by bvaughn** [comment]
I'm not sure why you would need to copy paste the same component. You should be able to import it. Let's say your project structure looked like this:
```
.
└── src
    ├── PageOne
    │   ├── index.js
    │   └── PageOne.js
    ├── PageTwo
    │   ├── index.js
    │   └── PageTwo.js
    └── SharedComponent
        ├── index.js
        └── SharedComponent.js
```
If both pages (`PageOne` and `PageTwo`) want to use the same, shared component (`SharedComponent`) then they could import it like so:

```js
// src/PageOne/PageOne.js
import SharedComponent from '../SharedComponent'; // Relative import
// You can now use <SharedComponent /> in your render method in PageOne
```

The specifics vary some depending on how your project is structured. You might want to post on Reactiflux or Stack Overflow if you have more questions about this though. 😄

**Comment by jvitoroc** [comment]
I didn't realize that before, maybe it's because my shared component is too small and would be unnecessary create it's own folder. Anyways, I did what you suggested and thank you.

**Comment by gaearon** [comment]
You don't need to create a folder for it. You can just keep it as a file, and then import it from different folder. I think it's not exactly clear what the problem was in your original question.

**Comment by jvitoroc** [comment]
Yes, I don't need to, but it would be inconsistent. When I started to build this app I couldn't scale it (it hurts to remember), I was using the [React/Redux file structure](https://marmelab.com/blog/2015/12/17/react-directory-structure.html) even without using Redux in my project. After converting to component folder pattern (explained in [this article](https://medium.com/styled-components/component-folder-pattern-ee42df37ec68)), I thought that each folder (aka feature, component) had to be exported to **src/index.js** and nothing else. Importing a feature inside another feature would be a bad practice. So I came up with this question.

Yes, my english is terrible.
